pennylane/devices/default_qubit.py

```diff
@@ -154,8 +154,10 @@ class DefaultQubit(QubitDevice):
         "Hamiltonian",
     }
 
-    def __init__(self, wires, *, shots=None, cache=0, analytic=None):
-        super().__init__(wires, shots, cache=cache, analytic=analytic)
+    def __init__(self, wires, *, shots=None, cache=0, analytic=None, custom_decomps=None):
+        super().__init__(
+            wires, shots, cache=cache, analytic=analytic, custom_decomps=custom_decomps
+        )
```

Comment:
If `custom_decomps` is used with external devices (that do not pass it to `super`), will nothing happen? Is this a problem?

Reply:
Ooh this gives me an idea :) In the _device loader_, you could do

```python
def device(name, *args, **kwargs):
    custom_decomp = kwargs.pop("custom_decomp", None)

    dev = ... # load the device

    if custom_decomp is not None:
      custom_decomp_expand_fn = qml.transforms.create_custom_decomp_expand_fn(
          custom_decomps, dev
      )
      dev.custom_expand(custom_decomp_expand_fn)
 
    ...
```
and the subclasses never have to worry :)

Reply:
> with decomposition_context(dev, custom_decomps): 
    print(qml.draw(circuit, expansion_strategy="device")(weights))

This is like what was implemented in #1872 . 

>Since dev.custom_expand will globally change the state of the device, I am always concerned about having that be user-facing without wrapping it in a context manager (which provides a context where the change of state is explicit).

In this case though, if the argument is passed upon creation of the device, I would expect the behaviour to be global, and that any time that device is used, the custom decomposition is used.

I do really like the idea of putting this in the device loader though; way more flexible, plus we were going to put the docs there anyways.